hydra/hydra.go

```diff
@@ -192,7 +205,73 @@ func NewHydra(ctx context.Context, options Options) (*Hydra, error) {
 		go handleBootstrapStatus(hdCtx, bsCh)
 
 		hds = append(hds, hd)
+
+		// dial back hooks
+		evts := []interface{}{
+			new(event.EvtPeerIdentificationCompleted),
+		}
+
+		subs, err := hd.Host.EventBus().Subscribe(evts, eventbus.BufSize(256))
+		if err != nil {
+			return nil, fmt.Errorf("head could not subscribe to eventbus events; err: %w", err)
+		}
+		go func(hd *head.Head, subs event.Subscription) {
+			defer subs.Close()
+			for {
+				select {
+				case evt := <-subs.Out():
+					ev := evt.(event.EvtPeerIdentificationCompleted)
+					p := ev.Peer
+					addrs := hd.Host.Peerstore().Addrs(p)
+
+					// dial back on quic if peer advertises a quic address
```

Comment:
It would be good to somehow exclude inbound dials from other hydras. Although I don't think that's going to be possible... but you could probably exclude heads from the same hydra.

Reply:
Oh you mean, if the inbound dial is from a head in our hydra, don't attempt this dial back ?

Reply:
yes

Reply:
If we're waiting for identify to complete we could check the useragent and exclude hydras

Reply:
@aschmahmann I agree. Will put in a fix.